wgpu-native/src/instance.rs

```diff
@@ -415,7 +428,8 @@ pub fn request_adapter(
                 adapter,
                 &mut token,
             );
-            return pick(id_out, id_vulkan);
+            pick(Some(id_out), id_vulkan);
```

Comment:
can we make it so the `pick` has to be called on returning? otherwise we have an implicit requirement that they always come together

Reply:
Sure, I'll update it

Reply:
@kvark I wrapped all of the `return`s into a closure so we can `assert` before the outer function returns. Does this look alright or did you have something else in mind?